### PR TITLE
Added #9 largest-non-adjacent-sum

### PR DESCRIPTION
**Pull request by NandanSatheesh**
largest-non-adjacent-sum was solved using Dynamic Programming approach.

### CODE REVIEW

File: challenges/largest-non-adjacent-sum.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import math
+
+def maximumNonAdjacentSum(array):
+    
+    DP = [0]*len(array)
+    
+    DP[0] = array[0]
+    DP[1] = max(array[0] , array[1])
+
```
Comment (seeni-dev) [inline challenges/largest-non-adjacent-sum.py:31]: What happens when len(array)==1 ?

Comment (seeni-dev) [reply challenges/largest-non-adjacent-sum.py:31]: and also it fails when len(array)==0.

File: challenges/largest-non-adjacent-sum.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+# since we pick 5 and 5.
+# 
+# 
+
+import math
+
+def maximumNonAdjacentSum(array):
+    
+    DP = [0]*len(array)
+
```
Comment (seeni-dev) [inline challenges/largest-non-adjacent-sum.py:27]: space is O(n). It is not a constant space. If you look closely on the algorithm, you are using only three at any instance, DP_i, DP_i_1, DP_i_2. You can rewrite the alogithm to do that

**Review by seeni-dev — CHANGES_REQUESTED** [review]

File: challenges/largest-non-adjacent-sum.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    
+    DP = [0]*len(array)
+    
+    DP[0] = array[0]
+    DP[1] = max(array[0] , array[1])
+    
+    for i in range(2,len(array)):
+    	
+        DP[i] = max3( array[i] , array[i] + DP[i-2] , DP[i-1])
+        print(DP)
```
Comment (seeni-dev) [inline challenges/largest-non-adjacent-sum.py:35]: as pointed out above rewrite this line to use constant space

**Comment by NandanSatheesh** [comment]
Will make the necessary changes and raise a new PR again. Thanks for the feedback.

**Pull request closed without merging**